packages/mobx-state-tree/src/core/mst-operations.ts

```diff
@@ -729,7 +729,7 @@ export function clone<T extends IAnyStateTreeNode>(
     return node.type.create(
         node.snapshot,
         keepEnvironment === true
-            ? node.root.environment
+            ? node.environment
```

Comment:
we can use environment from the node directly since they properly have the same environment as their parent all the way to the root
that way we don't need to traverse the tree all the way to the root